JavaScript/packages/recognizers-number-with-unit/src/numberWithUnit/models.ts

```diff
@@ -21,48 +21,56 @@ export abstract class AbstractNumberWithUnitModel implements IModel {
         query = QueryProcessor.preProcess(query, true);
 
         let extractionResults = new Array<ModelResult>();
-        for (let kv of this.extractorParsersMap.entries()) {
-            let extractor = kv[0];
-            let parser = kv[1];
-            let extractResults = extractor.extract(query);
-            let parseResults: Array<ParseResult> = [];
-            for (let i = 0; i < extractResults.length; i++) {
-                let r = parser.parse(extractResults[i]);
-                if (r.value !== null) {
-                    if (r.value instanceof Array) {
-                        for (let j = 0; j < r.value.length; j++) {
-                            parseResults.push(r.value[j]);
+
+        try{
```

Comment:
Shouldn't this type of try/catch be in the extract/parser level to make sure individual entity processings don't blow?

Reply:
> Shouldn't this type of try/catch be in the extract/parser level to make sure individual entity processings don't blow?

This part of .NET does the same thing.